### PR TITLE
bump knative.dev/net-istio to v0.37.1 and knative.dev/serving to v0.37.2

### PR DESCRIPTION
**Pull request by cpanato**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -20,10 +20,10 @@ require (
 	k8s.io/client-go v0.26.0
 	knative.dev/caching v0.0.0-20230418150356-05a86ad5d7db
 	knative.dev/hack v0.0.0-20230417170854-f591fea109b3
-	knative.dev/net-istio v0.36.2
+	knative.dev/net-istio v0.36.3
```
Comment (cpanato) [inline go.mod:23]: should we bump to 0.37.1?

Comment (k4leung4) [reply go.mod:23]: please bump to 0.37.1

Comment (cpanato) [reply go.mod]: @k4leung4 done PTAL

**Review by k4leung4 — APPROVED** [review]
thanks

**Pull request merged**